NGitLab/Models/SnippetCreateFile.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System.ComponentModel.DataAnnotations;
+using System.Runtime.Serialization;
+using System.Text.Json.Serialization;
+
+namespace NGitLab.Models
+{
+    public class SnippetCreateFile
+    {
+        public enum ActionType
```

Comment:
This enum and the `PreviousFile` property are only allowed to [update a snippet](https://docs.gitlab.com/ee/api/snippets.html#update-snippet). So, it should not be part of the `SnippetCreateFile` class.

The `Create` method uses the `POST` verb, whereas the `Update` (non-existant in NGitLab now) uses `PUT`. So, we cannot use the same method to create and update a snippet. 
I think, we should add a new method to update a snippet to the client, and create a new class `SnippetUpdate` class with this enum and other properties. This could be part of another PR if you prefer.

Reply:
Ok, yes this will be a bigger change ! ;)
I will try to do it today or tomorrow.